File: SumofLeftLeaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        if(!root) return 0;
        return (root->left&&!root->left->right&&!root->left->left?root->left->val:0)+sumOfLeftLeaves(root->right)+sumOfLeftLeaves(root->left);
    }
};

File: SumofLeftLeaves.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if(not root): return 0
        if(root.left and not root.left.left and not root.left.right):
            return root.left.val+Solution.sumOfLeftLeaves(self,root.left)+Solution.sumOfLeftLeaves(self,root.right)
        else:
            return Solution.sumOfLeftLeaves(self,root.left)+Solution.sumOfLeftLeaves(self,root.right)
